web3/middleware/stalecheck.py

```diff
@@ -0,0 +1,55 @@
+import datetime
+import time
+
+
+SKIP_STALECHECK_FOR_METHODS = set([
+    'eth_getBlockByHash',
+    'eth_getBlockByNumber',
+    'eth_getBlockTransactionCountByHash',
+    'eth_getBlockTransactionCountByNumber',
+    'eth_getTransactionByBlockHashAndIndex',
+    'eth_getTransactionByBlockNumberAndIndex',
+    'eth_getUncleCountByBlockHash',
+    'eth_getUncleCountByBlockNumber',
+])
+
+
+def _isfresh(block, allowable_delay):
+    return block and time.time() - block.timestamp <= allowable_delay
```

Comment:
Can we change this to use `block['timestamp']`.  I think it might be a bad idea for us to rely on the `AttributeDict` wrapper in our tests.

Reply:
Sure, are you thinking for flexibility to move back away from `AttributeDict` in the future? Or because an `AttributeDict` failure would cause test failures in unrelated places?